dvc/stage/run.py

```diff
@@ -159,17 +154,31 @@ def _checkpoint_run(stage, callback_func, done_cond, proc):
                 logger.exception(
                     "Error generating checkpoint, %s will be aborted", stage
                 )
-                proc.terminate()
+                _kill(proc)
+                killed.set()
             finally:
                 logger.debug("Remove checkpoint signal file")
                 os.remove(signal_path)
-        if done_cond.wait(1):
+        if done.wait(1):
             return
 
 
+def _kill(proc):
+    if os.name == "nt":
+        return _kill_nt(proc)
+    proc.terminate()
+    proc.wait()
+
+
+def _kill_nt(proc):
+    # windows stages are spawned with shell=True, proc is the shell process and
+    # not the actual stage process - we have to kill the entire tree
+    subprocess.call(["taskkill", "/F", "/T", "/PID", str(proc.pid)])
```

Comment:
This is something that we could consider doing via `psutil`, which has the logic needed to actually walk Windows process trees and find each child process individually (native python does not provide an easy way to do that). psutil has Windows wheels for <3.9 but still does not have any wheels for linux/mac, so if we were going to re-introduce it as a dependency it would have to be Windows only.

But calling `taskkill` here also works for our purposes